src/pretix/api/views/order.py

```diff
@@ -138,6 +141,16 @@ def subevent_after_qs(self, qs, name, value):
             )
             return qs
 
+        def provider_qs(self, qs, name, value):
+            qs = qs.annotate(
+                has_payment_with_provider=Exists(
+                    OrderPayment.objects.filter(
+                        Q(order=OuterRef('pk')) & Q(provider=value)
+                    )
+                )
+            ).filter(has_payment_with_provider=True)
+            return qs
```

Comment:
That looks a bit complicated to me as we anyway filter for true, ie. only existing payments. So for me it is easier to read:

```suggestion
            return qs.filter(Exists(
                OrderPayment.objects.filter(order=OuterRef('pk'), provider=value)
            ))
```